### ISSUE TITLE
Multiselect

### ISSUE THREAD

**Issue by tillprochaska** [k-next/plugins#7]
While most Javascript dropdown/select libraries provide a lot of additional, advanced functionality, many lack proper keyboard and screenreader support. Also, they are usually quite big in file size. That’s why I’ve started implementing a custom select using the new Custom Elements and Shadow DOM standards with a focus on accesibility and API compatibility with native select elements. 

Eventually, I’d like to create a simple multi select Panel field based on this. But this will probably still take a few weeks. Maybe someone else is already working on something like this or knows about a lightweight alternative?

**Comment by bvdputte** [comment]
I need something too.

I firmly believe we should use the already existing "Tags Field UI" ([demo](https://codepen.io/bastianallgeier/pen/OZyXoM)) for it, since it is a "core field" so it will be good with keyboard support etc and it will also use a design consistent with the rest of the panel. But then we'ld need a way to
- Feed it with a predefined set of options (same as you'ld do with the "select field" e.g.)
- Disable the adding of new items (hence make it select-only)

Bastian already said he'ld make a screencast on extending core fields, so I'm waiting on that first.

IMHO this would be a core field and the tags field would be an extension on this ;-)

**Comment by tillprochaska** [comment]
That’s a neat idea!

**Comment by distantnative** [comment]
Just little insight for you: the tags field already supports the same options functionality as the select field (in the form of autocomplete).

**Comment by olach** [comment]
I will port my sortable multiselect plugin to Kirby 3.
https://github.com/olach/kirby-relationship

But since this plugin takes up a lot of ui space in the panel, an alternative multiselect field in the form of a classic select dropdown would be a great complement if ui space is important.

**Comment by distantnative** [comment]
In v2 I made one of the first multiselect fields (https://github.com/distantnative/field-multiselect):
What is your take on this? Would it made sense for v3? Time has passed, maybe it is outdated and another approach needed? 

I know the v2 plugin has been used a lot. So I don't want to leave users without any equivalent option in v3. But given my time constraints I would also be fine with discontinuing it, if someone else has a better (or as good) plugin for v3.

**Comment by texnixe** [comment]
@distantnative As @olach mentioned, from a functional point of view, the Kirby-relationsship plugin is a good replacement, but I can imagine that within structures or tables, it would be nice to have your multiselect field.

**Comment by tillprochaska** [comment]
Hey @distantnative! It’s great to hear your working on updating the field. When working with large datasets, I think a select field/combo box is the only ui that really works well with limited space. As stated above, my primary concern is support for keyboard shortcuts and other accessibility features.

My time’s currently very limited as well and I’m not primarily working on a plugin like this, but I’d be happy to contribute wherever I can. (I’ve done some research on ux patterns and accessibility recommendations around custom select fields.)

I’d also like to check out @bvdputte approach and see if the tags field can be used for this, but this will still have to wait.

**Comment by distantnative** [comment]
@tillprochaska I am already quite far with rewriting it.
I'd be super interested in your recommendations for keyboard shortcuts and other accessibility features. Currently there aren't any so far.

**Comment by distantnative** [comment]
The Multiselect field has been added to the core.

I will copy your comments over as a ticket to improve the accessibility of `k-dropdown` in general.
